### ISSUE TITLE
Crash (NullPointerException) when adding new device

### ISSUE THREAD

**Issue by precurse** [syncthing/syncthing-android#834]
### Version Information

    App Version: 0.9.3
    Syncthing Version: v0.14.16
    Android Version: 6.0.1 Android patch level: Dec 1, 2016  (OnePlus Two)

This was happening since I started a fresh install of Syncthing on my phone and other devices.  Simply adding a new device causes the GUI to crash. At times the remote Syncthing instance will notify that I'm trying to add it (and appear on my phone after Syncthing is restarted),  but most of the time the GUI just silently crashes.  I had to pull the log from adb logcat:

```
Process: com.nutomic.syncthingandroid
Flags: 0x38c83e44
Package: com.nutomic.syncthingandroid v4113 (0.9.3)
Build: OnePlus/OnePlus2/OnePlus2:6.0.1/MMB29M/1447841100:user/release-keys

java.lang.NullPointerException: Attempt to read from field 'boolean com.nutomic.syncthingandroid.model.Connections$Connection.connected' on a null object reference
	at com.nutomic.syncthingandroid.views.DevicesAdapter.getView(DevicesAdapter.java:48)
	at android.widget.AbsListView.obtainView(AbsListView.java:2348)
	at android.widget.ListView.makeAndAddView(ListView.java:1876)
	at android.widget.ListView.fillSpecific(ListView.java:1355)
	at android.widget.ListView.layoutChildren(ListView.java:1663)
	at android.widget.AbsListView.onLayout(AbsListView.java:2150)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.support.v4.view.ViewPager.onLayout(ViewPager.java:1795)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.support.v4.widget.DrawerLayout.onLayout(DrawerLayout.java:1193)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2707)
	at android.view.View.layout(View.java:16692)
	at android.view.ViewGroup.layout(ViewGroup.java:5449)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2178)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1938)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1114)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6068)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
	at android.view.Choreographer.doCallbacks(Choreographer.java:670)
	at android.view.Choreographer.doFrame(Choreographer.java:606)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:152)
	at android.app.ActivityThread.main(ActivityThread.java:5497)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

```

**Comment by precurse** [comment]
I should note that adding it through the `Web GUI` portion of the app seems to work.

**Comment by Nutomic** [comment]
This was already fixed in 0780d606d974ad037e556f28e0bfb22fe94c12d2.
